src/actions/ui.js

```diff
@@ -78,10 +78,17 @@ export function showSource(sourceId: string) {
 }
 
 export function togglePaneCollapse(position: string, paneCollapsed: boolean) {
-  return {
-    type: "TOGGLE_PANE",
-    position,
-    paneCollapsed
+  return ({ dispatch, getState }: ThunkArgs) => {
+    const getPaneCollapseState = getPaneCollapse(getState(), position);
+    if (getPaneCollapseState === paneCollapsed) {
+      return;
+    }
+
+    dispatch({
+      type: "TOGGLE_PANE",
+      position,
+      paneCollapsed
+    });
```

Comment:
this is a good opportunity to update the tests in `actions/test/ui.spec.js`

Reply:
Sounds like a good idea, I'll take a look